src/liteclient.cc

```diff
@@ -93,50 +93,60 @@ LiteClient::LiteClient(Config& config_in, const AppEngine::Ptr& app_engine, bool
   } else {
     throw std::runtime_error("Unsupported package manager type: " + config.pacman.type);
   }
-  if (finalize) {
-    finalizeInstallation();
-  } else {
+
+  if (!finalize) {
     const auto current_target = getCurrent();
     update_request_headers(http_client, current_target, config.pacman);
     writeCurrentTarget(current_target);
+  } else if (!finalizeInstall()) {
+    LOG_ERROR << "Finalize Install failed";
   }
 }
 
-bool LiteClient::finalizeInstallation() {
-  data::InstallationResult update_finalization_result{data::ResultCode::Numeric::kNeedCompletion, ""};
-
-  // check if there is a pending update/installation/Target
-  boost::optional<Uptane::Target> pending_target;
-  storage->loadInstalledVersions("", nullptr, &pending_target);
-
-  // finalize a pending update/installation if any
-  if (!!pending_target) {
-    LOG_INFO << "Finalizing installation of the pending Target: " << pending_target->filename()
-             << ", hash: " << pending_target->sha256Hash();
-    // if there is a pending update/installation/Target then try to apply/finalize it
-    update_finalization_result = package_manager_->finalizeInstall(*pending_target);
-    if (update_finalization_result.isSuccess()) {
-      LOG_INFO << "Marking target install complete for: " << pending_target->filename();
-      storage->saveInstalledVersion("", *pending_target, InstalledVersionUpdateMode::kCurrent);
-    } else if (data::ResultCode::Numeric::kInstallFailed == update_finalization_result.result_code.num_code) {
-      LOG_ERROR << "Failed to finalize the installation, rollback has happened";
-      // rollback has happenned, unset is_pending and was_installed flags of the given Target record in DB
-      storage->saveInstalledVersion("", *pending_target, InstalledVersionUpdateMode::kNone);
+data::InstallationResult LiteClient::finalizePendingUpdate(boost::optional<Uptane::Target>& target) {
+  data::InstallationResult ret{data::ResultCode::Numeric::kNeedCompletion, ""};
+  InstalledVersionUpdateMode mode = InstalledVersionUpdateMode::kPending;
+  bool rollback = false;
+  LOG_INFO << "Finalize Pending, Target: " << target->filename() << ", hash: " << target->sha256Hash();
+
+  ret = package_manager_->finalizeInstall(*target);
+  if (ret.isSuccess()) {
+    LOG_INFO << "Finalize Complete, Target: " << target->filename();
+    mode = InstalledVersionUpdateMode::kCurrent;
+  } else {
+    rollback = data::ResultCode::Numeric::kInstallFailed == ret.result_code.num_code;
```

Comment:
I wanted to clearly identify the rollback sequence with a boolean (I think it helps in terms of semantics)

Reply:
ok. Will check it with details later.

Reply:
sounds good...this constructor seems to be doing way to much stuff, more than is required to instantiate a LiteClient class.